ptf/saiacl.py

```diff
@@ -0,0 +1,6068 @@
+
+        sai_thrift_remove_route_entry(self.client, self.route_entry)
+        sai_thrift_remove_next_hop(self.client, self.nhop)
+        sai_thrift_remove_neighbor_entry(self.client, self.neighbor_entry)
+
+        super(TCPFlagsACLTest, self).tearDown()
+
+
+@group("draft")
+class AclTableTypeTest(SaiHelper):
+    '''
+    ACL Type class. This test creates tables with various match fields
+    '''
+    acl_range_type = sai_thrift_s32_list_t(count=0, int32list=[])
+
+    def setUp(self):
+        super(AclTableTypeTest, self).setUp()
+
+        self.dmac = '00:11:22:33:44:55'
+        self.src_mac = '00:22:22:22:22:22'
+        self.ip_addr1 = '10.0.0.1'
+        self.ip_addr2 = '10.10.10.2'
+
+        self.port24_rif = sai_thrift_create_router_interface(
+            self.client,
+            type=SAI_ROUTER_INTERFACE_TYPE_PORT,
+            virtual_router_id=self.default_vrf,
+            port_id=self.port24)
+
+        self.port25_rif = sai_thrift_create_router_interface(
+            self.client,
+            type=SAI_ROUTER_INTERFACE_TYPE_PORT,
+            virtual_router_id=self.default_vrf,
+            port_id=self.port25)
+
+        self.nhop1 = sai_thrift_create_next_hop(
+            self.client,
+            ip=sai_ipaddress(self.ip_addr2),
+            router_interface_id=self.port25_rif,
+            type=SAI_NEXT_HOP_TYPE_IP)
+        self.neighbor_entry1 = sai_thrift_neighbor_entry_t(
+            rif_id=self.port25_rif, ip_address=sai_ipaddress(self.ip_addr2))
+        sai_thrift_create_neighbor_entry(
+            self.client, self.neighbor_entry1, dst_mac_address=self.dmac)
+
+        self.route_entry0 = sai_thrift_route_entry_t(
+            vr_id=self.default_vrf, destination=sai_ipprefix('10.0.0.1/32'))
+        sai_thrift_create_route_entry(
+            self.client, self.route_entry0, next_hop_id=self.nhop1)
+
+        self.route_entry1 = sai_thrift_route_entry_t(
+            vr_id=self.default_vrf,
+            destination=sai_ipprefix(
+                '1234:5678:9abc:def0:4422:1133:5577:99aa/128'))
+        sai_thrift_create_route_entry(
+            self.client, self.route_entry1, next_hop_id=self.nhop1)
+
+    def runTest(self):
+        self.testIPv4Acl()
+        self.testIPv6Acl()
+        self.testIPMirrorAcl()
+
+    def tearDown(self):
+        sai_thrift_remove_route_entry(self.client, self.route_entry0)
+        sai_thrift_remove_route_entry(self.client, self.route_entry1)
+        sai_thrift_remove_next_hop(self.client, self.nhop1)
+        sai_thrift_remove_neighbor_entry(self.client, self.neighbor_entry1)
+        sai_thrift_remove_router_interface(self.client, self.port25_rif)
+        sai_thrift_remove_router_interface(self.client, self.port24_rif)
+        super(AclTableTypeTest, self).tearDown()
+
+    def testIPv4Acl(self):
+        '''
+        Verify various IPv4 field combinations for table creation
+        '''
+        print("testIPv4Acl")
+        pkt = simple_tcp_packet(eth_dst=ROUTER_MAC,
```

Comment:
V4/V6 ACLs are applicable for L2 bridged traffic as well. CAn we include some tests with L2 switched traffic?

Reply:
In all the tests, eth_dst is router_mac. This will do routing based on the fib entry in v4 or v6 fib table. But these V4/V6 ACLs are also used for switched traffic.

For eg: please check FdbStaticMacTest in saifdb.py. Based on the MAC entry in the mac table, packets are forwarded to the egress port or flood if there is no MAC entry in the table. This happens when macda is not a router_mac. You can use those as examples for switched traffic